database/worker.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package database
+
+import (
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GetWorker gets a worker by hostname from the database.
+func (c *client) GetWorker(hostname string) (*library.Worker, error) {
+	logrus.Tracef("Getting worker %s from the database", hostname)
+
+	// variable to store query results
+	w := new(database.Worker)
+
+	// send query to the database and store result in variable
+	err := c.Database.
+		Table(constants.TableWorker).
+		Raw(c.DML.WorkerService.Select["worker"], hostname).
+		Scan(w).Error
+
+	return w.ToLibrary(), err
+}
+
+// GetWorker gets a worker by hostname from the database.
+func (c *client) GetWorkerByAddress(address string) (*library.Worker, error) {
+	logrus.Tracef("Getting worker %s from the database", address)
+
+	// variable to store query results
+	w := new(database.Worker)
+
+	// send query to the database and store result in variable
+	err := c.Database.
+		Table(constants.TableWorker).
+		Raw(c.DML.WorkerService.Select["address"], address).
+		Scan(w).Error
+
+	return w.ToLibrary(), err
+}
+
+// GetWorkerList gets a list of all workers from the database.
+func (c *client) GetWorkerList() ([]*library.Worker, error) {
+	logrus.Trace("Listing workers from the database")
+
+	// variable to store query results
+	w := new([]database.Worker)
+
+	// send query to the database and store result in variable
+	err := c.Database.
+		Table(constants.TableWorker).
+		Raw(c.DML.WorkerService.List["all"]).
+		Scan(w).Error
+
+	// variable we want to return
+	workers := []*library.Worker{}
+	// iterate through all query results
+	for _, worker := range *w {
+		// https://golang.org/doc/faq#closures_and_goroutines
+		tmp := worker
+
+		// convert query result to library type
+		workers = append(workers, tmp.ToLibrary())
+	}
+
+	return workers, err
+}
+
+// GetWorkerCount gets a count of all workers from the database.
+func (c *client) GetWorkerCount() (int64, error) {
+	logrus.Trace("Counting workers in the database")
+
+	// variable to store query results
+	var w []int64
+
+	// send query to the database and store result in variable
+	err := c.Database.
+		Table(constants.TableWorker).
+		Raw(c.DML.WorkerService.Select["count"]).
+		Pluck("count", &w).Error
+
+	return w[0], err
```

Comment:
(i know it exists elsewhere in the codebase so feel free to resolve this)
But if the Database.Table.Raw.Pluck fails will (L87) cause a segfault? Should we check if len(w) == 0 ?

Reply:
I think we should check that len is 0 or that err != nil. I'll make that change